osu.Game/Screens/OnlinePlay/Multiplayer/Multiplayer.cs

```diff
@@ -35,16 +37,32 @@ private void onRoomUpdated()
                 transitionFromResults();
         }
 
+        private void onLoadAborted()
+        {
+            // If the server aborts gameplay for this user (due to loading too slow), exit gameplay screens.
+            if (!this.IsCurrentScreen())
+            {
+                Logger.Log("Gameplay aborted because loading the beatmap took too long.", LoggingTarget.Runtime, LogLevel.Important);
+                this.MakeCurrent();
```

Comment:
Seems safe for now, but we had better hope that we don't cause any exit blocking operations in the future.